src/core/pdf-parser/parseName.ts

```diff
@@ -35,8 +35,12 @@ const parseName = (
   if (!result) return undefined;
 
   const [fullMatch, name] = result;
+  const decoded = name.replace(
```

Comment:
I think this decoding is a good idea. I'm not sure if the parsing function is the best spot to do it though. What do you think about moving this logic into the constructor of the [`PDFName`](https://github.com/Hopding/pdf-lib/blob/master/src/core/pdf-objects/PDFName.ts#L15) object itself? This object already has to encode characters in its [`toString()`](https://github.com/Hopding/pdf-lib/blob/master/src/core/pdf-objects/PDFName.ts#L45-L55) method. So I think it makes sense to have the decoding logic in there too.

Reply:
I had originally wondered about that... I guess I wasn't sure if having the encoding in the `.toString()` is actually ideal. Maybe better would be to make a `PDFName.fromEncoded()` as there is already a `PDFName.from()` in use. Thoughts?

Reply:
Sure, creating a `PDFName.fromEncoded()` and placing the decoding logic there makes sense to me. That way it's possible to create a `PDFName` object without going through the decoding process, if needed.